src/python/gudhi/off_utils.pyx

```diff
@@ -25,21 +26,95 @@ __license__ = "MIT"
 cdef extern from "Off_reader_interface.h" namespace "Gudhi":
     vector[vector[double]] read_points_from_OFF_file(string off_file)
 
+def _get_next_line(file_desc, comment='#'):
+    """Return the next line that is not a comment.
+
+    :param file_desc: An open file in read mode.
+    :type file_desc: file
+    :param comment: The characters or list of characters used to indicate the start of a comment.
+    :type comment: string
+
+    :returns: The next line.
+    :rtype: string
+    """
+    while True:
+        # file_desc.readline() is preferred to next(file_desc), as the second option is not compatible with
+        # seek and tell methods
+        line = file_desc.readline()
+        if not line.startswith(comment):
+            break
+    return line
+
+def _read_off_file_header(file_desc):
+    """Return the information contained in the header of an OFF file.
+
+    :param file_desc: An open file in read mode.
+    :type file_desc: file
+
+    :returns: The point cloud dimension, and the number of points.
+    :rtype: tuple(int, int)
+
+    Raises:
+        ValueError: If the file does not respect the OFF file format.
+    """
+    nb_vertices = -1
+
+    line = _get_next_line(file_desc)
+    # First line should be "OFF" (3d case) or "nOFF" (dD case)
+    if line.lower().startswith("off"):
+        dim = 3
+    elif line.lower().startswith("4off"):
+        dim = 4
+    elif line.lower().startswith("noff"):
```

Comment:
In the C++ code, we are a bit more flexible and accept things like COFF (see http://www.geomview.org/docs/html/OFF.html). I don't know if that's important.

Reply:
Ok, let's consider '*OFF' that is not 'nOFF', nor '4OFF' as a 3d OFF file, just the C++ code is doing. Changed on [9732176](https://github.com/GUDHI/gudhi-devel/pull/1008/commits/973217672dad0e842d200a042d7eeaf85329fe58)